test/test_dataloader.py

```diff
@@ -2266,6 +2272,22 @@ def test_excessive_thread_creation_warning(self):
             dataloader = DataLoader(self.dataset, batch_size=2, num_workers=1000)
 
 
+class TestDataLoaderDeviceType(TestCase):
+    def test_nested_tensor_multiprocessing(self, device):
+        dataset = [torch.nested.nested_tensor([torch.randn(5)], device=device) for _ in range(100)]
+
+        loader = torch.utils.data.DataLoader(
+            dataset,
+            batch_size=1,
+            num_workers=4,
+            collate_fn=_identity,
```

Comment:
Sadly no, there is some dense-specific logic that does concatenation directly into shared memory. NT doesn't support `nt.new(storage)` at the moment:
```
RuntimeError: Unrecognized tensor type ID: NestedTensorCUDA
```

Alternatively, we could add nested support to the default `collate_fn`?

Reply:
Conceptually, there's two ways I can see the default `collate_fn` being expanded:
1. **Input batch list contains dense tensors of ragged sizes**: produce a NT output instead of `torch.stack()`ing. This is a potentially disruptive change so it may be better to require an explicit `collate_fn` override to do this at first. We could throw a descriptive error that points out the raggedness and suggests passing a `collate_fn` that returns a NT.
2. **Input batch list contains NTs** : should the default behavior be to concatenate the list of NTs into a dim+1 NT? This is the case described in the linked issue, although they're implementing a `collate_fn` that just returns the list directly.